### PR TITLE
Refs Task#381528 #381564 update webhook

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#381528](https://dev.sun-asterisk.com/issues/381528)
- [#381564](https://dev.sun-asterisk.com/issues/381564)
## What's this PR do ?

- [x] show and update webhook

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [ ] It was tested in local success?
- [ ] Fill link PR into ticket and the opposite
- [ ] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
ảnh giao diện : https://drive.google.com/file/d/1XKZ_FNTPh1yWsPFYuoL_ODoOLMqala10/view?usp=sharing
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr></table>

File: app/Repositories/Eloquents/WebhookRepository.php [hunk 1]
```diff
@@ -20,4 +20,8 @@ public function getAllByUser()
                             ->orderBy('webhooks.created_at', 'desc')
                             ->get();
     }
+
+    public function getWebhookByName($name) {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/WebhookRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/WebhookRepository.php</br>**Line**: 24</br>Opening brace should be on a new line

File: app/Http/Controllers/WebhookController.php [hunk 2]
```diff
@@ -83,9 +88,22 @@ public function edit(Webhook $webhook)
      * @param  \App\Models\Webhook  $webhook
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, Webhook $webhook)
+    public function update(WebhookUpdateRequest $request)
     {
-        //
+        $data = $request->except('_token');
+        $check_webhook = $this->webhookRepository->getWebhookByName($data['name']);
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 94</br>Variable "check_webhook" is not in valid camel caps format

File: app/Http/Controllers/WebhookController.php [hunk 3]
```diff
@@ -83,9 +88,22 @@ public function edit(Webhook $webhook)
...
+        if(isset($check_webhook)) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 95</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 95</br>Variable "check_webhook" is not in valid camel caps format

File: app/Http/Controllers/WebhookController.php [hunk 4]
```diff
@@ -83,9 +88,22 @@ public function edit(Webhook $webhook)
...
+            if($check_webhook->id != $data['id']){
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 96</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 96</br>Variable "check_webhook" is not in valid camel caps format

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 96</br>Expected 1 space(s) after closing parenthesis; found 0

File: app/Http/Controllers/WebhookController.php [hunk 5]
```diff
@@ -83,9 +88,22 @@ public function edit(Webhook $webhook)
...
+                return redirect()->back()->with('messageFailName', 'This webhook name has already been used by another webhook')->withInput();
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Line exceeds 120 characters; contains 142 characters

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr></table>

File: app/Repositories/Eloquents/WebhookRepository.php (hunk 1 above)
Comment (framgiaciv3) [inline app/Repositories/Eloquents/WebhookRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/WebhookRepository.php</br>**Line**: 24</br>Opening brace should be on a new line

File: app/Http/Controllers/WebhookController.php [hunk 6]
```diff
@@ -83,9 +89,23 @@ public function edit(Webhook $webhook)
      * @param  \App\Models\Webhook  $webhook
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, Webhook $webhook)
+    public function update(WebhookUpdateRequest $request)
     {
-        //
+        $data = $request->except('_token');
+        $check_webhook = $this->webhookRepository->getWebhookByName($data['name']);
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 95</br>Variable "check_webhook" is not in valid camel caps format

File: app/Http/Controllers/WebhookController.php [hunk 7]
```diff
@@ -83,9 +89,23 @@ public function edit(Webhook $webhook)
...
+        if(isset($check_webhook)) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 96</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 96</br>Variable "check_webhook" is not in valid camel caps format

File: app/Http/Controllers/WebhookController.php [hunk 8]
```diff
@@ -83,9 +89,23 @@ public function edit(Webhook $webhook)
...
+            if($check_webhook->id != $data['id']){
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Variable "check_webhook" is not in valid camel caps format

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Expected 1 space(s) after closing parenthesis; found 0

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Blank line found at start of control structure

File: app/Http/Controllers/WebhookController.php [hunk 9]
```diff
@@ -83,9 +89,23 @@ public function edit(Webhook $webhook)
...
+                
+                return redirect()->back()->with('messageFailName', 'This webhook name has already been used by another webhook')->withInput();
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 99</br>Line exceeds 120 characters; contains 142 characters

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/WebhookController.php [hunk 10]
```diff
@@ -83,9 +89,26 @@ public function edit(Webhook $webhook)
      * @param  \App\Models\Webhook  $webhook
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, Webhook $webhook)
+    public function update(WebhookUpdateRequest $request)
     {
-        //
+        $data = $request->except('_token');
+        $response = $this->webhookRepository->getWebhookByName($data['name']);
+        if(isset($response)) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 96</br>Expected 1 space(s) after IF keyword; 0 found

File: app/Http/Controllers/WebhookController.php [hunk 11]
```diff
@@ -83,9 +89,26 @@ public function edit(Webhook $webhook)
...
+            if($response->id != $data['id']){
```
Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterKeyword</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Expected 1 space(s) after IF keyword; 0 found

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Expected 1 space(s) after closing parenthesis; found 0

Comment (framgiaciv3) [inline app/Http/Controllers/WebhookController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/WebhookController.php</br>**Line**: 97</br>Blank line found at start of control structure

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>failure</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
OK

File: app/Http/Controllers/WebhookController.php [hunk 12]
```diff
@@ -71,9 +72,14 @@ public function store(WebhookCreateRequest $request)
      * @param  \App\Models\Webhook  $webhook
      * @return \Illuminate\Http\Response
      */
-    public function edit(Webhook $webhook)
+    public function edit($id)
     {
-        return view('webhooks.edit');
+        $webhook = $this->webhookRepository->find($id);
```
Comment (tamtt-1031) [inline app/Http/Controllers/WebhookController.php]: trường hợp webhook không phải của thằng current user đang đăng nhập thì nó chỉnh url vẫn xem được thông tin đúng ko em?
ví dụ:
Current User: có webhook->id = 1 
Another User: có webhook->id = 2

current user nó edit url thành http://127.0.0.1:8000/bots/2/edit thì nó vẫn xem được webhook của thằng Another User.

Anh nghĩ là nên query thêm theo điều kiệu user_id

File: app/Http/Requests/WebhookUpdateRequest.php [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+            'room_id' => 'required',
+        ];
+    }
+    public function messages()
+    {
+        return [
+            'name.required' => 'Please enter name',
+            'name.min' => 'Name is too short (minimum is 5 characters)',
+            'name.max' => 'Name is too long (maximum is 200 characters)',
+            'name.unique' => 'This webhook name has already been used by another webhook',
```
Comment (tamtt-1031) [inline app/Http/Requests/WebhookUpdateRequest.php:44]: user khác tạo cùng name rồi có được tạo lại ko em?
a nghĩ validate unique name theo thằng user_id thôi e nhỉ.

Comment (nguyenthinhhn) [reply app/Http/Requests/WebhookUpdateRequest.php:44]: trường hợp này em check trong controller rồi a, không được trùng tên webhook nào hết,

File: app/Http/Controllers/WebhookController.php (hunk 12 above)
Comment (nguyenthinhhn) [reply app/Http/Controllers/WebhookController.php]: ok a

File: app/Http/Controllers/WebhookController.php [hunk 14]
```diff
@@ -83,9 +89,25 @@ public function edit(Webhook $webhook)
...
+    public function update(WebhookUpdateRequest $request)
     {
-        //
+        $data = $request->except('_token');
+        $response = $this->webhookRepository->getWebhookByName($data['name']);
+        if (isset($response)) {
+            if ($response->id != $data['id']) {
+                return redirect()
+                    ->back()
+                    ->with('messageFailName', 'This webhook name has already been used by another webhook')
```
Comment (ghost) [inline app/Http/Controllers/WebhookController.php]: đoạn này validate trùng tên webhook theo user phải ko e.
nếu vậy thì e nên validate ở webhookUpdateRequest luôn nhé

File: app/Http/Requests/WebhookUpdateRequest.php (hunk 13 above)
Comment (tamtt-1031) [reply app/Http/Requests/WebhookUpdateRequest.php:44]: a nghĩ check trùng theo từng thằng user thôi, còn thằng user khác có thể tạo lại với tên đó được

Comment (nguyenthinhhn) [reply app/Http/Requests/WebhookUpdateRequest.php:44]: như thế thì admin nhìn sẽ thấy những tên trùng, mình để check tất luôn thì k có trường hợp nào trùng tên hết,

File: app/Http/Controllers/WebhookController.php (hunk 14 above)
Comment (nguyenthinhhn) [reply app/Http/Controllers/WebhookController.php]: update mà validate thì nó sẽ check cả trùng bản ghi hiện tại luôn a, nên phải bỏ sang controler check trường hợp không thay đổi tên bàn ghi hien tại

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Http/Controllers/WebhookController.php [hunk 15]
```diff
@@ -73,7 +75,14 @@ public function store(WebhookCreateRequest $request)
      */
     public function edit(Webhook $webhook)
     {
-        return view('webhooks.edit');
+        if (Auth::user()->role == UserType::USER && $webhook->user_id != Auth::id()) {
+            abort(404, 'Page not found');
```
Comment (ghost) [inline app/Http/Controllers/WebhookController.php]: lỗi này là lỗi 403 - permission denied chứ, với cái này e dùng policy đi https://laravel.com/docs/5.7/authorization#generating-policies , mà phần này của task authorization mà

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

File: app/Http/Requests/WebhookUpdateRequest.php (hunk 13 above)
Comment (ghost) [inline app/Http/Requests/WebhookUpdateRequest.php:44]: em xem dùng cách này thử nè https://laracasts.com/discuss/channels/requests/problem-with-unique-field-validation-on-update

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

File: app/Http/Requests/WebhookUpdateRequest.php (hunk 13 above)
Comment (tamtt-1031) [reply app/Http/Requests/WebhookUpdateRequest.php:44]: @thinhbnv-0938 admin có thể xem của user nào. a nghĩ nên theo hương người dùng thôi. còn admin chỉ là quản lý thôi e.
VD: a có 2 user khác nhau có 2 webhook đều nhưng đều về mảng tin tức và muốn đặt tên webhook của mình là "Tin tức" ko lẽ user đầu tiên đặt rồi user thứ 2 lại ko đc đặt tên đó. :D

File: resources/views/webhooks/edit.blade.php [hunk 16]
```diff
@@ -1,4 +1,138 @@
...
+		</div>
+		<div class="col-xs-8">
+			<label class="field-compulsory required" for="webhook_name">Webhook name</label>
+			<input type="text" id="webhook_name" name="name" class="form-control" value="{{  $webhook->name }}">
+			@error('name')
+			<div class="has-error">
+				<span class="help-block">{{ $message }}</span>
+			</div>
+			@enderror
+			@if(Session::has('messageFailName'))
```
Comment (ghost) [inline resources/views/webhooks/edit.blade.php]: xóa cái này luôn đi e

Comment (nguyenthinhhn) [reply resources/views/webhooks/edit.blade.php]: yep

File: app/Http/Requests/WebhookUpdateRequest.php (hunk 13 above)
Comment (nguyenthinhhn) [reply app/Http/Requests/WebhookUpdateRequest.php:44]: ok, để e sửa theo yc của a

**Review by ghost — APPROVED** [review]

File: app/Policies/WebhookPolicy.php [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+     * @return mixed
+     */
+    public function before($user, $ability)
+    {
+        if ($user->role == UserType::ADMIN) {
+            return true;
+        }
+    }
+
+    public function viewAny(User $user)
```
Comment (ghost) [inline app/Policies/WebhookPolicy.php]: hàm này này không dùng cho nó bay màu đi e :v

File: app/Policies/WebhookPolicy.php [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    /**
+     * Determine whether the user can view the webhook.
+     *
+     * @param  \App\Models\User  $user
+     * @param  \App\Webhook  $webhook
+     * @return mixed
+     */
+    public function view(User $user, Webhook $webhook)
```
Comment (ghost) [inline app/Policies/WebhookPolicy.php]: hàm này nữa

File: app/Policies/WebhookPolicy.php [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+        return true;
+    }
+
+    /**
+     * Determine whether the user can create webhooks.
+     *
+     * @param  \App\Models\User  $user
+     * @return mixed
+     */
+    public function create(User $user)
```
Comment (ghost) [inline app/Policies/WebhookPolicy.php]: hàm này nữa

File: app/Policies/WebhookPolicy.php [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    /**
+     * Determine whether the user can restore the webhook.
+     *
+     * @param  \App\Models\User  $user
+     * @param  \App\Webhook  $webhook
+     * @return mixed
+     */
+    public function restore(User $user, Webhook $webhook)
```
Comment (ghost) [inline app/Policies/WebhookPolicy.php]: hàm này nữa

File: app/Policies/WebhookPolicy.php [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    /**
+     * Determine whether the user can permanently delete the webhook.
+     *
+     * @param  \App\Models\User  $user
+     * @param  \App\Webhook  $webhook
+     * @return mixed
+     */
+    public function forceDelete(User $user, Webhook $webhook)
```
Comment (ghost) [inline app/Policies/WebhookPolicy.php]: hàm này nữa

Comment (nguyenthinhhn) [reply app/Policies/WebhookPolicy.php]: ok . e xóa hết r

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

File: app/Http/Requests/WebhookUpdateRequest.php [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'name' => 'required|min:1|max:50',
+            'description' => 'required|min:10|max:1000|',
+            'bot_id' => 'required',
+            'room_name' => 'required',
```
Comment (framgiaciv3) [inline app/Http/Requests/WebhookUpdateRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/WebhookUpdateRequest.php</br>**Line**: 33</br>Each line in an array declaration must end in a comma

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by tamtt-1031 — APPROVED** [review]

**Review by quanghung97 — APPROVED** [review]
OK

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**